### PR TITLE
Upgrade to supp lvl

### PR DESCRIPTION
**Pull request by Armatorix**
Summary of changes:
- use of snap plugin library
- update tests
- add licenses

How to verify it:
- make test-all

Testing done:
- make test-all

A picture of a snapping turtle (not required but encouraged):
https://i.ytimg.com/vi/LBSFeJ1WLv4/maxresdefault.jpg

### CODE REVIEW

File: iostat/iostat.go [hunk 1]
```diff
@@ -29,63 +29,61 @@ import (
 
 	"github.com/intelsdi-x/snap-plugin-collector-iostat/iostat/command"
 	"github.com/intelsdi-x/snap-plugin-collector-iostat/iostat/parser"
-	"github.com/intelsdi-x/snap/control/plugin"
-	"github.com/intelsdi-x/snap/control/plugin/cpolicy"
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
 	"github.com/intelsdi-x/snap/core"
```
Comment (kdembler) [inline iostat/iostat.go]: Plugin using lib-go shouldn't import snap

File: glide.lock [hunk 2]
```diff
@@ -1,5 +1,5 @@
 hash: e2b8455d2f47652e91d0350ad7994f4824c6728af81e95c454094734bccbba32
```
Comment (kdembler) [inline glide.lock]: `glide.lock` needs to be updated to reflect changes to dependencies: not importing snap, importing lib-go. I suggest removing both `glide.lock`, `glide.yaml`, clearing glide cache with `$ glide cc` and then going through `$ glide init` to ensure proper dependency list.

File: iostat/iostat.go [hunk 3]
```diff
@@ -29,63 +29,61 @@ import (
...
 
 const (
 	// Name of plugin
 	Name = "iostat"
-	// Version of plugin
-	Version = 6
-	// Type of plugin
-	Type         = plugin.CollectorPluginType
+	//Version of plugin
+	Version      = 6
```
Comment (kdembler) [inline iostat/iostat.go]: Please bump up version

File: main.go [hunk 4]
```diff
@@ -20,33 +20,25 @@ limitations under the License.
...
 
-	// Import the snap plugin library
 	"github.com/intelsdi-x/snap-plugin-collector-iostat/iostat"
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
+)
 
-	// Import our collector plugin implementation
-	"github.com/intelsdi-x/snap/control/plugin"
+const (
+	pluginName    = "iostat"
```
Comment (kdembler) [inline main.go]: Please choose one place to store version and name of plugin - here in `main.go` or in `iostat/iostat.go` and remove the other.

**Review by kdembler — CHANGES_REQUESTED** [review]

File: main.go (hunk 4 above)
Comment (Armatorix) [inline main.go]: Already deleted from iostat.go.

File: iostat/iostat.go [hunk 5]
```diff
@@ -172,21 +163,26 @@ func (iostat *IOSTAT) GetMetricTypes(_ plugin.ConfigType) ([]plugin.MetricType,
...
+func (iostat *Iostat) run(mts []plugin.Metric) ([]string, map[string]float64, error) {
 	// TODO: allow the path and/or name of the command to be overriden through the pluginConfigType
 
+	versionString := iostat.cmd.Exec("iostat", []string{"-V"})
+	version, err := iostat.parser.ParseVersion(versionString)
+	if err != nil {
+		return nil, nil, err
+	}
+	if version[0] < 10 || (version[0] == 10 && version[1] < 2) {
+		return nil, nil, fmt.Errorf("Iostat %d.%d.%d version (required >=10.2.0)", version[0], version[1], version[2])
```
Comment (kdembler) [inline iostat/iostat.go]: This error message could be a little bit more intuitive (e.g. `This plugin requires iostat in version 10.2.0 or newer (version present={version})`)

File: scripts/deps.sh [hunk 6]
```diff
@@ -60,7 +60,7 @@ restore_go_dep() {
       (cd "${__proj_dir}" && godep restore)
       ;;
     glide)
-      (cd "${__proj_dir}" && glide install)
+      (cd "${__proj_dir}"  && glide install)
```
Comment (kdembler) [inline scripts/deps.sh]: Please remove space

**Review by kdembler — APPROVED** [review]

**Pull request merged**